pkg/clustersync/clusterSync.go

```diff
@@ -314,4 +326,52 @@ func processClusterDelete(ctx context.Context, obj interface{}) {
 		return
 	}
 	dao.DeleteClusterAndResources(ctx, clusterName, deleteClusterNode)
+
+}
+
+var managedClusterResourceGvr = schema.GroupVersionResource{
+	Group:    "cluster.open-cluster-management.io",
+	Version:  "v1",
+	Resource: "managedclusters",
+}
+
+// finds lingering data in database from deleted/detached clusters or clusters with search-collector-addon disabled:
+func confirmDelete(ctx context.Context) ([]string, error) {
+	var managedClustersFromClient, needToDelete []string
+
+	managedClustersFromDB, _ := dao.GetManagedCluster(ctx)
+	klog.V(5).Infof("Managed Clusters reported from database: ", managedClustersFromDB)
+
+	// get all managed clusters via dynamic client:
+	resourceObj, err := config.GetDynamicClient().Resource(managedClusterResourceGvr).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		klog.Warning("Error resolving ManagedClusters with dynamic client", err.Error())
+		return nil, err
+	}
+	for _, item := range resourceObj.Items {
+
+		// Here we want all managed clusters that have the search-collector addon available
+		// TO DO: need better method instead of using name.
+		if item.GetName() != "local-cluster" &&
+			item.GetLabels()["feature.open-cluster-management.io/addon-search-collector"] == "available" {
+			managedClustersFromClient = append(managedClustersFromClient, item.GetName())
+		}
+	}
+	klog.V(5).Infof("Managed Clusters reported from kube client: ", managedClustersFromClient)
+
+	// Iterate managed clusters reported by kube client and managed clusters with delete event.
+	// If find match we push remaining cluser to delete func
+	for _, mClusters := range managedClustersFromClient {
+		for _, dmClusters := range managedClustersFromDB {
+			if mClusters != dmClusters {
+				needToDelete = append(needToDelete, dmClusters)
+				klog.V(3).Infof("Found Managed Cluster data in database that should be deleted! Cluster found: %s", dmClusters)
+			} else {
+				klog.V(5).Infof("Managed Cluster data successfully deleted from database.")
+			}
+		}
+	}
```

Comment:
This block isn't working correctly.
I'd change managedClustersFromClient to a `map[string]struct{}`
Then you can change this loop to...
```go
for _, dmCluster := range managedClustersFromDB {
	if _, exists := managedClusterFromClient[dmCluster]; !exist {
		// At this point the cluster exists in DB, but not in the list from client.
		needToDelete = append(needToDelete, dmCluster)
		klog.V(3).Infof("Found Managed Cluster data in database that should be deleted! Cluster found: %s", dmClusters)
		} else {
			klog.V(5).Infof("Managed Cluster data successfully deleted from database.")
		}
	}
}
```